File: workspace1/week2/src/com/hackbulgaria/programming51/week2/CoffeeMachine.java
//Coffee Machine
//In a com.hackbulgaria.programming51.week2 package create a CoffeeMachine class,
//where you have the following attributes:
//
//Coffee[] brandsOfCoffee - an array for Coffee. One coffee machine can hold up to 10 brands of coffee.
//float balance - the current money balance in the coffee machine.
//And the following methods:
//
//Create method for adding Coffee in the machine - public boolean addCoffee(Coffee coffee). 
//The method should return true if you can add that coffee type to the machine. 
//Otherwise, return false (more than 10 cups are added)
//Create method for buying Coffee from the Coffee Machine public float buyCoffee(String brand, float price).
//This method recieves coffee brand and money given by the buyer. 
//If the coffee brand exsists in brandsOfCoffee - return the change. 
//If it doesn't - return -1. The amounts of coffee we can buy, for now, is unlimited.
//Example in main:
//
//CoffeeMachine cm = new CoffeeMachine();
//Coffee java = new Coffee();
//Coffee latte = new Coffee();
//
//java.brand = "Java";
//java.price = 1.00;
//
//latte.brand = "Latte";
//latte.price = 1.50;
//
//cm.addCoffee(java);
//cm.addCoffee(latte);
//
//System.out.println(cm.buyCoffee("Latte",3.40));
//System.out.println("Current balance: " + cm.balance));
//Should return:
//
//1.90
//Current balance: 1.50


package com.hackbulgaria.programming51.week2;

public class CoffeeMachine {
	
	private Coffee [] brands = new Coffee[10];
	private float currentBalans = 0;
	private int currentIndex = 0;
	private int maxCapacity = 10;
	private float reminder = 0;
	private float pay = 0;
	//Default string
	public CoffeeMachine() {
		
	}
	
	public boolean addToBrands(String brand) {
		if (currentIndex < maxCapacity) {
			brands[currentIndex] = brand;
			currentIndex++;
			return true;
		}
		return false;
	}
	
	private float givingChange(float pay, float price) {
		reminder = pay - price;
		return reminder;
	}
	
	public float machineBalans(String brand, float pay) {
		for (int i = 0; i < 10; i ++) {
			if (brands[i].equals(brand)) {
				
			}
		}
	}

}

File: C Tasks/coffeeMachine.c
/*Coffee Machine*/
/*Coffee[] brandsOfCoffee - an array for Coffee. One coffee machine can hold up to 10 brands of coffee.*/
/*float balance - the current money balance in the coffee machine.*/
/*And the following methods:*/

/*Create method for adding Coffee in the machine - public boolean addCoffee(Coffee coffee). The method should return true if you can add that coffee type to the machine. Otherwise, return false (more than 10 cups are added)*/
/*Create method for buying Coffee from the Coffee Machine public float buyCoffee(String brand, float price). This method recieves coffee brand and money given by the buyer. If the coffee brand exsists in brandsOfCoffee - return the change. If it doesn't - return -1. The amounts of coffee we can buy, for now, is unlimited.*/

#include <stdio.h>
#include <string.h>

int capacity = 0;
int total = 0;

typedef struct coffee {
    char brand[10];
    float price;
} Coffee;

typedef struct coffeeMachine {
    Coffee coffee[5];
    float balance;
} machine;

main()
{
    machine myMachine;
    
    myMachine.balance = 0;
    
    int i;
    char name[10];
    float price;
    float givingchange;
    float pay;
    int choose;
    int c;
    
    for (i = 0; i < 5; i++) {
        printf("\nEnter brand of coffee: ");
        scanf("%9s", name);
        printf("Enter price for coffee: ");
        scanf("%f", &price);
        
        strcpy(myMachine.coffee[i].brand, name);
        myMachine.coffee[i].price = price;
    }
    
    printf("\nChoose coffee:\n");
    for (i = 0; i < 5; i++) {
        printf("%d: %s\t%.2f\n", i + 1, myMachine.coffee[i].brand, myMachine.coffee[i].price);
    }

    while ((c = getchar()) != EOF) {
        printf("\nInsert your money: ");
        scanf("%f", &pay);
        
        choose = c - '0';
        
/*        printf("\nYour choise is: ");*/
/*        scanf("%d", &choose);*/
        
        switch (choose) {
            case 1: case 2: case 3: case 4: case 5:
                if (pay < myMachine.coffee[choose - 1].price) {
                    printf("\nNot enought money.\n");
                }
                else {
                    givingchange = pay - myMachine.coffee[choose - 1].price;
                    myMachine.balance += myMachine.coffee[choose - 1].price;
                    
                    printf("Your change: %.2f\n", givingchange);
                    printf("Balance: %.2f\n", myMachine.balance);
                }
                break;
                
            default:
                printf("\nWrong choise.\nTry again.\n");
                break;
        }
    }
}
